pkgs/tools/networking/n2n/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, autoconf, automake, pkg-config }:
+
+stdenv.mkDerivation rec {
+  pname = "n2n";
+  version = "2.8";
+
+  src = fetchFromGitHub {
+    owner = "ntop";
+    repo = "n2n";
+    rev = version;
+    hash = "sha256-2xJ8gYVZJZoKs6PZ/9GacgxQ+/3tmnRntT1AbPe1At4=";
+  };
+
+  nativeBuildInputs = [ autoconf automake pkg-config ];
+
+  preConfigure = ''
+    patchShebangs autogen.sh
+    ./autogen.sh
+  '';
```

Comment:
Could we replace this with `autoreconfHook`?

Reply:
I don't think so, but admittedly I barely know anything about how autotools work. It looks like `autogen.sh` adds version information to the configure script and then runs `autoreconf`. Since we have to patch `autogen.sh` anyway, I think just running it afterwards is shorter and clearer than adding a magic hook.

Reply:
> than adding a magic hook.

The hook is not magic, it is a shortcut.